runtime/ts/storage/firebase-storage.ts

```diff
@@ -557,6 +560,7 @@ class FirebaseVariable extends FirebaseStorageProvider {
   }
 
   async modelForSynchronization() {
+    await this.initialized;
```

Comment:
Bonus fix. The changes to _persistChanges altered the timing of setup, which exposed that we weren't awaiting initialization here ... which caused us to fall back to the default modelForSynchronization (because value is null before initialization) ... which would have done the write thing ... except that it (correctly) waits for synchronization, then ignores referenceMode.